capten/common-pkg/plugins/argocd/repositories.go

```diff
@@ -16,6 +16,12 @@ func (a *ArgoCDClient) CreateRepository(ctx context.Context, repo *Repository) (
 	}
 	defer io.Close(conn)
 
+	var repoUpsert bool
+	existingRepo, err := a.GetRepository(ctx, repo.Repo)
+	if existingRepo != nil && err == nil {
+		repoUpsert = true
+	}
+
```

Comment:
check existingRepo, match repo  that it retuined proper value.
first check err !=nil then check existingRepo